lib/ansible/modules/remote_management/redfish/redfish.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017-2018 Dell EMC Inc.
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+---
+module: redfish
+version_added: "2.7"
+short_description: Out-Of-Band management using Redfish APIs
+description:
+  - Builds Redfish URIs locally and sends them to remote OOB controllers to
+    get information back or perform an action.
+  - When receiving information, it is placed in location specified by user.
+options:
+  category:
+    required: true
+    description:
+      - Action category to execute on server
```

Comment:
Suggest adding `choices` to `category`, even if not to `command`

Reply:
Done in https://github.com/ansible/ansible/pull/41297/commits/5edc431e08450555660efff5b57a3978b380801c